File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.go
func findMaxAverage(nums []int, k int) float64 {
    var currentVal float64=0
    temp:=math.Inf(-1)
    for i:=0; i<len(nums); i++{
        if i<k{
            currentVal+=float64(nums[i])
        } else{
            if temp<currentVal{temp=currentVal}
            currentVal+=float64(nums[i]-nums[i-k])
        }
    }
    if temp<currentVal{temp=currentVal}
    return temp/float64(k)
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double currentVal=0, temp=INT_MIN;
        for(int i=0;i<nums.size();i++){
            if(i<k){currentVal+=nums[i];}
            else{
                temp=temp>currentVal?temp:currentVal;
                currentVal+=nums[i]-nums[i-k];
            }
        }
        temp=currentVal>temp?currentVal:temp;
        return temp/k;
    }
};